api/app/src/models/medical/organization.ts

```diff
@@ -1,41 +1,46 @@
 import { DataTypes, Sequelize } from "sequelize";
+import { Config } from "../../shared/config";
 import { OIDNode, OID_ID_START } from "../../shared/oid";
 import { BaseModel, defaultModelOptions, ModelSetup } from "../_default";
 
 export class Organization extends BaseModel<Organization> {
   static NAME = "organization";
-  declare id: number;
-  declare systemRootOid: string;
+  declare id: string;
+  declare cxId: string;
+  declare organizationId: number;
+  declare data: object;
 
   static setup: ModelSetup = (sequelize: Sequelize) => {
     Organization.init(
       {
         ...BaseModel.baseAttributes(),
         id: {
-          type: DataTypes.INTEGER,
-          primaryKey: true,
-        },
-        systemRootOid: {
           type: DataTypes.STRING,
           primaryKey: true,
         },
+        cxId: {
+          type: DataTypes.UUID,
+        },
+        organizationId: {
+          type: DataTypes.INTEGER,
+          unique: true,
+        },
+        data: {
+          type: DataTypes.JSONB,
+        },
       },
       {
         ...defaultModelOptions(sequelize),
         tableName: Organization.NAME,
         hooks: {
           async beforeCreate(attributes) {
-            const curMaxId = (await Organization.max("id", {
-              where: { systemRootOid: attributes.systemRootOid },
-            })) as number;
-            attributes.id = curMaxId ? (attributes.id = curMaxId + 1) : OID_ID_START;
+            const curMaxId = (await Organization.max("organizationId")) as number;
+            const orgId = curMaxId ? curMaxId + 1 : OID_ID_START;
+            attributes.id = `${Config.getSystemRootOID()}.${OIDNode.organizations}.${orgId}`;
+            attributes.organizationId = orgId;
```

Comment:
I get the concern about multiple things being created at the same time, but if the concern is a race condition - we should be validating this with table locking behavior, and not retries/recursive functions.

Can you elaborate on what you mean here?

Reply:
I'm assuming there's no locking to generate this ID, since this is done in memory and I couldn't find anything that would indicate a semaphore/lock.

Retrying is, in theory, cheaper than locking the whole table or building some other type of lock to prevent more than one resource with the same ID.

I took a quick look at what Postgres have for this, [this](https://stackoverflow.com/a/13727025/2099911) seems like the best approach other than retrying.

Reply:
Note that we can use `SERIAL` here -> but only b/c org IDs need to be unique across cxs. This approach won't work for facilities/patients, so will need to have another solution here

Reply:
Will take care of this under https://github.com/metriport/metriport-internal/issues/366 before we ship the API to customers